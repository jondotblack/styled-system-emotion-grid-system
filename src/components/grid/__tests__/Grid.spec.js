import React from 'react';
import { render, screen, rendererWithTheme } from 'testing/utils';

import '@testing-library/jest-dom/extend-expect';

import { Col, Container, Row } from '../index';

describe('Grid', () => {
  it('should render a container correctly with all props', () => {
    const tree = rendererWithTheme(<Container fluid>container content</Container>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a row correctly', () => {
    const tree = rendererWithTheme(<Row>row content</Row>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render width styles correctly with span prop', () => {
    const tree = rendererWithTheme(
      <Col span={{ md: 2, lg: 3, xl: 4 }}>column content</Col>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render offset styles correctly with prop', () => {
    const tree = rendererWithTheme(<Col offset={{ md: 1, lg: 3 }}>column content</Col>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render order styles correctly with prop', () => {
    const tree = rendererWithTheme(<Col order={{ md: 1, lg: 2 }}>column content</Col>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('children elements are correctly rendering', () => {
    render(
      <Container>
        <Row>
          <Col>placeholder text</Col>
        </Row>
      </Container>,
    );

    const children = screen.getByText('placeholder text');

    expect(children).toBeInTheDocument();
  });
});
