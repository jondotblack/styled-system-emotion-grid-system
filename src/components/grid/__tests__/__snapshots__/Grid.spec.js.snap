// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render a container correctly with all props 1`] = `
.emotion-0 {
  width: 100%;
  max-width: 1400px;
  margin-right: auto;
  margin-left: auto;
  padding-right: 1rem;
  padding-left: 1rem;
  max-width: 100%;
}

<div
  className="emotion-0 emotion-1"
>
  container content
</div>
`;

exports[`Grid should render a row correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 100%;
  -ms-flex: 1 0 100%;
  flex: 1 0 100%;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-right: calc(24px / -2);
  margin-left: calc(24px / -2);
}

<div
  className="emotion-0 emotion-1"
>
  row content
</div>
`;

exports[`Grid should render offset styles correctly with prop 1`] = `
.emotion-0 {
  position: relative;
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  width: 100%;
  max-width: 100%;
  padding-right: calc(24px / 2);
  padding-left: calc(24px / 2);
}

@media screen and (min-width:768px) {
  .emotion-0 {
    margin-left: 8.333333333333332%;
  }
}

@media screen and (min-width:992px) {
  .emotion-0 {
    margin-left: 25%;
  }
}

<div
  className="emotion-0 emotion-1"
  order={null}
>
  column content
</div>
`;

exports[`Grid should render order styles correctly with prop 1`] = `
.emotion-0 {
  position: relative;
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  width: 100%;
  max-width: 100%;
  padding-right: calc(24px / 2);
  padding-left: calc(24px / 2);
}

@media screen and (min-width:768px) {
  .emotion-0 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media screen and (min-width:992px) {
  .emotion-0 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media screen and (min-width:992px) {
  .emotion-0 {
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

<div
  className="emotion-0 emotion-1"
  order={
    Object {
      "lg": 2,
      "md": 1,
    }
  }
>
  column content
</div>
`;

exports[`Grid should render width styles correctly with span prop 1`] = `
.emotion-0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  position: relative;
  -webkit-flex: 1 0 0%;
  -ms-flex: 1 0 0%;
  flex: 1 0 0%;
  width: 100%;
  max-width: 100%;
  padding-right: calc(24px / 2);
  padding-left: calc(24px / 2);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

@media screen and (min-width:768px) {
  .emotion-0 {
    width: 16.666666666666664%;
  }
}

@media screen and (min-width:992px) {
  .emotion-0 {
    width: 25%;
  }
}

@media screen and (min-width:1200px) {
  .emotion-0 {
    width: 33.33333333333333%;
  }
}

@media screen and (min-width:768px) {
  .emotion-0 {
    width: 16.666666666666664%;
  }
}

@media screen and (min-width:992px) {
  .emotion-0 {
    width: 25%;
  }
}

@media screen and (min-width:1200px) {
  .emotion-0 {
    width: 33.33333333333333%;
  }
}

<div
  className="emotion-0 emotion-1"
  order={null}
  width={
    Object {
      "lg": "25%",
      "md": "16.666666666666664%",
      "xl": "33.33333333333333%",
    }
  }
>
  column content
</div>
`;
